File: Compiled/CPP/timer.cpp
#include <iostream>
#include <iomanip>
#include <Windows.h>
#include <cmath>

int main()
{
    int t;
    std::cout << "Enter a time in seconds: ";
    std::cin >> t;

    while (t)
    {
        int mins = floor(t / 60);
        int secs = t % 60;

        std::cout << std::setfill('0') << std::setw(2) << mins;
        std::cout << ":";
        std::cout << std::setfill('0') << std::setw(2) << secs;
        std::cout << "\r";

        Sleep(1000);
        t -= 1;
    }
    return 0;
}

File: Compiled/Nim/timer.nim
import std/math, strutils, strformat
import os

write(stdout, "Enter the time in seconds: ")

var s = readLine(stdin)
var t = parseInt(s)

while t != 0:
    var mins = math.floor(t / 60)
    var secs = t mod 60

    var str = &"{mins:02}:{secs:02}\r"
    write(stdout, str)

    os.sleep(1000)
    t -= 1